plugins/repository-hdfs/build.gradle

```diff
@@ -0,0 +1,184 @@
+      exclude module: "commons-beanutils"
+      exclude module: "commons-beanutils-core"
+      exclude module: "javax.servlet"
+      exclude module: "guava"
+      exclude module: "junit"
+      // provided by ES itself
+      exclude group: "org.slf4j"
+      exclude group: "log4j"
+  }
+
+  hadoop2("org.apache.hadoop:hadoop-hdfs:${versions.hadoop2}") {
+      exclude module: "guava"
+      exclude module: "log4j"
+      exclude module: "junit"
+      // provided by ES itself
+      exclude group: "org.slf4j"
+      exclude group: "log4j"
+  }
+} 
+
+configurations.all {
+    resolutionStrategy {
+        force "commons-codec:commons-codec:${versions.commonscodec}"
+        force "commons-logging:commons-logging:${versions.commonslogging}"
+        force "commons-lang:commons-lang:2.6"
+        //force "commons-collections:commons-collections:3.2.2"
+        force "commons-httpclient:commons-httpclient:3.0.1"
+        force "org.codehaus.jackson:jackson-core-asl:1.8.8"
+        force "org.codehaus.jackson:jackson-mapper-asl:1.8.8"
+    }
+}
+
+
+dependencyLicenses {
+  mapping from: /hadoop-.*/, to: 'hadoop'
+}
+
+compileJava.options.compilerArgs << '-Xlint:-deprecation,-rawtypes'
+
+// main jar includes just the plugin classes
+jar {
+    include "org/elasticsearch/plugin/hadoop/hdfs/*"
+}
+
+// haddop jar (which actually depend on Hadoop)
+task hadoopLinkedJar(type: Jar, dependsOn:jar) {
+    appendix "internal"
+    from sourceSets.main.output.classesDir
+    // exclude plugin
+    exclude "org/elasticsearch/plugin/hadoop/hdfs/*"
+}
+
+
+// configure 'bundle' as being w/o Hadoop deps
+bundlePlugin {
+    into ("internal-libs") {
+        from hadoopLinkedJar.archivePath
+    }
+}
+
+task distZipHadoop1(type: Zip, dependsOn: [hadoopLinkedJar, jar]) { zipTask ->
+    from zipTree(bundlePlugin.archivePath)
+    description = "Builds archive (with Hadoop1 dependencies) suitable for download page."
+    classifier = "hadoop1"
+
+    into ("hadoop-libs") {
+        from configurations.hadoop1.allArtifacts.files
+        from configurations.hadoop1
+    }
+}
+
+task distZipHadoop2(type: Zip, dependsOn: [hadoopLinkedJar, jar]) { zipTask ->
+    from zipTree(bundlePlugin.archivePath)
+    description = "Builds archive (with Hadoop2/YARN dependencies) suitable for download page."
+    classifier = "hadoop2"
+
+    into ("hadoop-libs") {
+        from configurations.hadoop2.allArtifacts.files
+        from configurations.hadoop2
+    }
+}
+
+artifacts {
+    archives bundlePlugin
+    'default' bundlePlugin
+    archives distZipHadoop1
+    archives distZipHadoop2
+}
+
+test {
+  // this is needed for insecure plugins, remove if possible!
+  //systemProperty 'tests.artifact', project.name 
+  testLogging {
+    outputMode = "ALWAYS"
+  }
+}
+
+integTest {
+    clusterConfig.plugin(pluginProperties.extension.name, zipTree(distZipHadoop1.archivePath))
+    logger.info("Installing plugin " + distZipHadoop1.archivePath)
+}
+
+integTest.enabled = false
```

Comment:
We should have a minimal rest test which just checks the plugin can be installed, like we do for all the other plugins. It is important we have at least this minimal level of checking that things work in a real situation (with a real elasticsearch setup, not mocked up test setup).